conda_package_streaming/package_streaming.py

```diff
@@ -30,13 +36,35 @@ def __str__(self):
 
 
 class TarfileNoSameOwner(tarfile.TarFile):
+    def __init__(self, *args, umask=umask, **kwargs):
+        """Open an (uncompressed) tar archive `name'. `mode' is either 'r' to
+        read from an existing archive, 'a' to append data to an existing
+        file or 'w' to create a new file overwriting an existing one. `mode'
+        defaults to 'r'.
+        If `fileobj' is given, it is used for reading or writing data. If it
+        can be determined, `mode' is overridden by `fileobj's mode.
+        `fileobj' is not closed, when TarFile is closed.
+        """
+        super().__init__(*args, **kwargs)
+        self.umask = umask
```

Comment:
There are a bunch of other constructors in the stdlib `TarFile` class (`taropen`, `open`, `gzopen`, `bz2open`, `xzopen`), do you think it would be worthwhile to wrap them as well, to make sure the `umask` is applied consistently?

Reply:
These other constructors invokes `__init__` and politely passes along `**kwargs` so AFAIK this is handled for us?

Reply:
This is exclusively for our particular usage of tar. No other code path should ever be relevant here.